sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,764 @@
+        default="ml.g5.4xlarge",
+    )
+    optional_args.add_argument(
+        "--train-on-cpu",
+        action="store_true",
+        help="Run training and inference on CPU instances instead of GPU.",
+    )
+    optional_args.add_argument(
+        "--graph-construction-instance-type",
+        type=str,
+        default=None,
+        help=(
+            "Instance type for graph construction. "
+            "Used in GSProcessing, GConstruct, GraphBoltConversion steps. "
+            "Default: same as CPU instance type"
+        ),
+    )
+    optional_args.add_argument(
+        "--gsprocessing-instance-count",
+        type=int,
+        default=None,
+        help="Number of GSProcessing instances. Default is equal to number of training instances.",
+    )
+    optional_args.add_argument(
+        "--volume-size-gb",
+        type=int,
+        help="Additional volume size for instances in GB.",
+        default=100,
+    )
+
+    # Pipeline/Task Configuration
+    required_args.add_argument(
+        "--graph-name", type=str, required=True, help="Name of the graph. Required."
+    )
+    required_args.add_argument(
+        "--input-data-s3",
+        type=str,
+        required=True,
+        help="S3 path to the input graph data. Required.",
+    )
+    required_args.add_argument(
+        "--output-prefix-s3",
+        type=str,
+        required=True,
+        help="S3 prefix for the output data. Required.",
+    )
+    optional_args.add_argument(
+        "--pipeline-name",
+        type=str,
+        help="Name for the pipeline. Needs to be unique per account and region.",
+    )
+    optional_args.add_argument(
+        "--base-job-name",
+        type=str,
+        default="gs",
+        help="Base job name for SageMaker jobs. Default: 'sm'",
+    )
+    required_args.add_argument(
+        "--jobs-to-run",
+        nargs="+",
+        required=True,
+        help=(
+            "Space-separated string of jobs to run in the pipeline, "
+            "e.g. 'gconstruct train inference'. "
+            f"Should be one or more of: {list(JOB_ORDER.keys())}. Required. "
+        ),
+    )
+    optional_args.add_argument(
+        "--log-level",
+        type=str,
+        default="INFO",
+        choices=["DEBUG", "INFO", "WARNING", "CRITICAL", "FATAL"],
+        help="Logging level for the jobs. Default: INFO",
+    )
+    optional_args.add_argument(
+        "--step-cache-expiration",
+        type=str,
+        default="30d",
+        help="Expiration time for the step cache, as a string in ISO 8601 duration. Default: 30d",
+    )
+    optional_args.add_argument(
+        "--update-pipeline",
+        action="store_true",
+        help="Update an existing pipeline instead of creating a new one.",
+    )
+
+    # Graph construction configuration
+    optional_args.add_argument(
+        "--graph-construction-config-filename",
+        type=str,
+        default="",
+        help="Filename for the graph construction config. "
```

Comment:
S3 url?

Reply:
It's only the filename here, we derive the full S3 path from the input prefix:

```
gc_proc_input = ProcessingInput(
            source=self.input_data_param, destination=gc_local_input_path
        )

[...]
gconstruct_arguments = [
            "--graph-config-path",
            self.graphconstruct_config_param,
            "--input-path",
            gc_local_input_path,
            ...
]
```